bot.user.js

```diff
@@ -259,26 +238,20 @@ var canvas = (function() {
         // Check if circles intersect
         circleIntersect: function(circle1, circle2) {
             var bothRadii = circle1.radius + circle2.radius;
-
-            // Pretends the circles are squares for a quick collision check.
-            // If it collides, do the more expensive circle check.
-            if (circle1.x + bothRadii > circle2.x &&
-                circle1.y + bothRadii > circle2.y &&
-                circle1.x < circle2.x + bothRadii &&
-                circle1.y < circle2.y + bothRadii) {
-
-                var distance = Math.sqrt(Math.pow(circle1.x - circle2.x, 2) +
-                    Math.pow(circle1.y - circle2.y, 2));
-                if (distance < bothRadii) {
-                    if (window.visualDebugging) {
-                        var collisionPointX = ((circle1.x * circle2.radius) + (circle2.x * circle1.radius)) / bothRadii;
-                        var collisionPointY = ((circle1.y * circle2.radius) + (circle2.y * circle1.radius)) / bothRadii;
-                        canvas.drawDot(collisionPointX, collisionPointY, circle2.radius, 'cyan', true);
-                        canvas.drawDot(circle2.x, circle2.y, circle2.radius, 'red', true);
-                    }
-                    return true;
-                }
-            }
+			var bothRadiiSqr = bothRadii*bothRadii;
+			var xDist = circle1.x - circle2.x;
+			var yDist = circle1.y - circle2.y;
+			var distance = xDist*xDist + yDist*yDist;
```

Comment:
At first I was going to say "don't we have a function for this?"

But now that I see [our function](https://github.com/ErmiyaEskandary/Slither.io-bot/blob/master/bot.user.js#L213-L245), maybe your way is better :worried: